packages/framework-core/src/decorators/projection.ts

```diff
@@ -35,4 +35,6 @@ function registerProjection(originName: string, projectionMetadata: ProjectionMe
   })
 }
 
-type ProjectionMethod<TEntity, TReadModel> = TypedPropertyDescriptor<(_: TEntity, readModel?: TReadModel) => TReadModel>
+type ProjectionMethod<TEntity, TReadModel> = TypedPropertyDescriptor<
+  (_: TEntity, readModel?: TReadModel) => ProjectionResult<TReadModel>
```

Comment:
Why not returning a `TReadModel | null` or a `Promise<TReadModel>` instead? The event dispatcher should have enough context to figure out which read model to delete, and we avoid introducing an extra interface to the public-facing SDK.

Reply:
Hey @javiertoledo there is no real advantage, it was just to make it more readable for the final user:
```
@ReadModel
export class UserReadModel {
  public constructor(readonly username: string, /* ...(other interesting fields from users)... */) {}
  
  @Projects(User, 'id')
  public static projectUser(entity: User, current?: UserReadModel): UserReadModel {
    if (....)) {
        return null // This could sound like "do nothing"
    }
    return new UserReadModel(...)
  }
```

But yeah it's true that the additional type is not needed at all, let me change it 👀

Reply:
Changed!

Reply:
I strongly disagree with returning `null`. The user can mess-up, return null by accident and delete production data. I don't think that we should allow this. Please reconsider this @javiertoledo @juanjoman